ow_lander/scripts/all_action_trajectories.py

```diff
@@ -0,0 +1,853 @@
+  """
+  :param approximate: use an approximate solution. default True
+  :type move_group: class 'moveit_commander.move_group.MoveGroupCommander'
+  :type x_start: float
+  :type y_start: float
+  :type z_start: float
+  :type approximate: bool
+  """
+  
+  move_arm.set_start_state(cs)
+
+  goal_pose.position.x = x_start
+  goal_pose.position.y = y_start
+  goal_pose.position.z = z_start
+
+  # Ask the planner to generate a plan to the approximate joint values generated
+  # by kinematics builtin IK solver. For more insight on this issue refer to:
+  # https://github.com/nasa/ow_simulator/pull/60
+  if approximate:
+    move_arm.set_joint_value_target(goal_pose, True)
+  else:
+    move_arm.set_pose_target(goal_pose)
+  plan = move_arm.plan()
+  if len(plan.joint_trajectory.points) == 0:  # If no plan found, abort
+    return False
+  return plan
+
+
+
+def dig_linear(move_arm, robot, moveit_fk, args):
+    
+  """
+  :type move_arm: class 'moveit_commander.move_group.MoveGroupCommander'
+  :type args: List[bool, float, int, float, float, float]
+  """
+  x_start = args.x_start
+  y_start = args.y_start
+  depth = args.depth
+  length = args.length
+  ground_position = args.ground_position
+  
+  plan_a= move_to_pre_trench_configuration(
+      move_arm, robot, x_start, y_start)
+  if len(plan_a.joint_trajectory.points) == 0:  # If no plan found, abort
+    return False
+
+  cs, start_state, current_pose = calculate_joint_state_end_pose_from_plan_arm (robot, plan_a, move_arm, moveit_fk)
+  #################### Rotate hand yaw to dig in#################################
+  plan_b = change_joint_value(move_arm, cs, start_state, constants.J_HAND_YAW, 0.0) 
+  
+  
+  if len(plan_b.joint_trajectory.points) == 0:  # If no plan found, send the previous plan only
+    return plan_a
+
+  dig_linear_traj = cascade_plans (plan_a, plan_b)
+  
+  ######################### rotate scoop #######################################
+
+  cs, start_state, current_pose = calculate_joint_state_end_pose_from_plan_arm (robot, dig_linear_traj, move_arm, moveit_fk)
+  plan_c = change_joint_value(move_arm, cs, start_state, constants.J_SCOOP_YAW, math.pi/2)
+  
+  dig_linear_traj = cascade_plans (dig_linear_traj, plan_c)
+  
+  ######################### rotate dist pith to pre-trenching position###########
+
+  cs, start_state, current_pose = calculate_joint_state_end_pose_from_plan_arm (robot, dig_linear_traj, move_arm, moveit_fk)
+  
+  plan_d = change_joint_value(move_arm, cs, start_state, constants.J_DIST_PITCH, -math.pi/2)
+
+  dig_linear_traj = cascade_plans (dig_linear_traj, plan_d)
+  
+  ##########################Once aligned to trench goal, 
+  ##########################place hand above the desired start point
+  alpha = math.atan2(constants.WRIST_SCOOP_PARAL, constants.WRIST_SCOOP_PERP)
+  distance_from_ground = constants.ROT_RADIUS * \
+      (math.cos(alpha) - math.sin(alpha))
```

Comment:
Indentation off.  Could be added to previous line.

Reply:
done!